.github/workflows/asv_benchmark_pr.yml

```diff
@@ -0,0 +1,72 @@
+name: Benchmark PR
+
+on:
+  pull_request:
+    branches: [main]
+
+permissions:
+  contents: read  # Read access for repository contents
+  pull-requests: write  # Write access for pull requests
+
+env:
+  PYTHON_VERSION: "3.10"
+  WORKING_DIR: ${{ github.workspace }}/benchmarks
+
+jobs:
+  benchmark-pr:
+    runs-on: ubuntu-latest
+
+    defaults:
+      run:
+        working-directory: ${{ env.WORKING_DIR }}
+
+    steps:
+
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ env.PYTHON_VERSION }}
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install asv virtualenv lf-asv-formatter
+
+    - name: Create ASV machine config file
+      run: asv machine --machine gh-runner --yes
+
+    - name: Save comparison of PR against main branch
+      run: |
+        # prepare main branch for comparison
+        git remote add upstream https://github.com/${{ github.repository }}.git
+        git fetch upstream main
+
+        # Run benchmarks, writing comment contents to ./output
+        asv continuous upstream/main HEAD \
```

Comment:
This maybe out of scope for this PR but just out of curiosity: It is possible to configure `asv` such that the machines on every run are somewhat similar. This makes it such that the benchmark comparision history is accurately built on the historical web browser window thing that `asv` has. (https://asv.readthedocs.io/en/stable/using.html#machine-information). 

Maybe that could also help out by allowing us to run the benchmarks only once per run and compare them with the existing `main` results which were stored as an artifact.

Reply:
I agree. We should create a separate issue which implements the Out of Scope section of the PRs description